### PR TITLE
[FAN-114] Add a detail jump to a single item in the brand list

### PR DESCRIPTION
**Pull request by XuMinhui**
Hello, brave developer, nice day to merge a pull request, ain't it?

## Follow the checklist to proceed

- [x] I've performed a code self-review
- [x] I've tested feature or bugfix on my own
- [x] If I make changes to another person's module, I've asked how to use it or request a review
- [x] I've checked mobile & tablet screens (if have changes in UI)
- [x] Local build successfully completed
- [x] Branch name and pull request title are according to [the requirements](https://fangible.atlassian.net/wiki/spaces/FAN/pages/62226433/Branches+naming)

### CODE REVIEW

File: src/modules/brand/actions/queryBrandPools.ts [hunk 1]
```diff
@@ -42,23 +50,82 @@ export const queryBrandPools = createSmartAction<RequestAction>(
...
+                  cts: new Array(tarTokenList.length).fill(
+                    params.contractaddress,
+                  ),
+                  ids: tarTokenList.map(item => item.tokenId),
                 }),
               );
-              return new Promise((resolve, reject) => resolve(itemData));
-            } else {
+
+              if (itemData?.length) {
```
Comment (0123cf) [inline src/modules/brand/actions/queryBrandPools.ts]: You don't need if

File: src/modules/brand/actions/queryBrandPools.ts [hunk 2]
```diff
@@ -42,23 +50,82 @@ export const queryBrandPools = createSmartAction<RequestAction>(
...
+                return itemData.map(item => {
+                  const findTar = tarTokenList.find(
```
Comment (0123cf) [inline src/modules/brand/actions/queryBrandPools.ts]: Better to use Map

File: src/modules/brand/actions/queryBrandPools.ts [hunk 3]
```diff
@@ -42,23 +50,82 @@ export const queryBrandPools = createSmartAction<RequestAction>(
...
+                    : {};
+                });
+              } else {
+                return [];
+              }
+            };
+
+            if (allToken.length === 0) {
               return new Promise((resolve, reject) => resolve([]));
+            } else {
```
Comment (0123cf) [inline src/modules/brand/actions/queryBrandPools.ts]: You don't need else

File: src/modules/brand/actions/queryBrandPools.ts [hunk 4]
```diff
@@ -42,23 +50,82 @@ export const queryBrandPools = createSmartAction<RequestAction>(
...
+              }
+
+              if (eaToken.length > 0) {
+                const queryEaList = querytarTokenList(eaToken);
+                PromiseList.push(queryEaList);
+              }
+
+              return Promise.all(PromiseList).then(res => {
+                const resArr: any[] | PromiseLike<any[]> = [];
+                for (let i = 0; i < res.length; i++) {
```
Comment (0123cf) [inline src/modules/brand/actions/queryBrandPools.ts]: You might need the flat function

```
then(res => res.flat())
```

**Review by 0123cf — APPROVED** [review]

**Review by 0123cf — APPROVED** [review]

**Review by 0123cf — APPROVED** [review]

File: src/modules/brand/screens/Brands/components/BrandNFTItems/BrandNFTItems.tsx [hunk 5]
```diff
@@ -32,6 +34,12 @@ interface IBrandNFTItemsProps {
   contractAddress: string;
 }
 
+interface ISwiperItem extends IItem {
+  auctionType?: AuctionType.FixedSwap;
```
Comment (Denise54321) [inline src/modules/brand/screens/Brands/components/BrandNFTItems/BrandNFTItems.tsx]: auctionType?: AuctionType

File: src/modules/brand/actions/queryBrandPools.ts [hunk 6]
```diff
@@ -42,24 +50,76 @@ export const queryBrandPools = createSmartAction<RequestAction>(
...
+              const queryFsList = querytarTokenList(fsToken);
+              PromiseList.push(queryFsList);
+            }
+
+            if (eaToken.length > 0) {
+              const queryEaList = querytarTokenList(eaToken);
+              PromiseList.push(queryEaList);
+            }
+
+            return Promise.all(PromiseList).then(res => res.flat());
```
Comment (Denise54321) [inline src/modules/brand/actions/queryBrandPools.ts:129]: querytarTokenList can be a separate method

**Review by Denise54321 — APPROVED** [review]

**Pull request merged**